rust-app/src/implementation.rs

```diff
@@ -131,39 +102,78 @@ pub static SIGN_IMPL: SignImplT = Action(
                                     _ if name == b"coin.TRANSFER" => { return None; }
-                                    _ => { return None; } // Change this to allow unknown capabilities.
+                                    Some((Some(Some(sender)), Some(Some(receiver)), Some(First(amount)), Some(Some(target_chain)))) if name == b"coin.TRANSFER_XCHAIN" => {
+                                        cross_transfer_prompt((sender, receiver, amount, target_chain));
+                                    }
+                                    Some((Some(Some(sender)), Some(Some(receiver)), Some(Second(Some(Decimal{field_decimal:Some (amount)}))), Some(Some(target_chain)))) if name == b"coin.TRANSFER_XCHAIN" => {
+                                        cross_transfer_prompt((sender, receiver, amount, target_chain));
+                                    }
+                                    _ if name == b"coin.TRANSFER_XCHAIN" => { return None; }
+                                    Some((None, None, None, None)) => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, no args", name_utf8)?))?;
+                                    }
+                                    Some((Some(Some(arg1)), None, None, None)) => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, arg 1: '{}'", name_utf8, from_utf8(arg1.as_slice())?)?))?;
+                                    }
+                                    Some((Some(Some(arg1)), Some(Some(arg2)), None, None)) => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, arg 1: '{}', arg 2: '{}'", name_utf8, from_utf8(arg1.as_slice())?, from_utf8(arg2.as_slice())?)?))?;
+                                    }
+                                    Some((Some(Some(arg1)), Some(Some(arg2)), Some(First(arg3)), None)) => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, arg 1: '{}', arg 2: '{}', arg 3: {}", name_utf8, from_utf8(arg1.as_slice())?, from_utf8(arg2.as_slice())?, from_utf8(arg3.as_slice())?)?))?;
+                                    }
+                                    Some((Some(Some(arg1)), Some(Some(arg2)), Some(First(arg3)), Some(Some(arg4)))) => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, arg 1: '{}', arg 2: '{}', arg 3: {}, arg 4: '{}'", name_utf8, from_utf8(arg1.as_slice())?, from_utf8(arg2.as_slice())?, from_utf8(arg3.as_slice())?, from_utf8(arg4.as_slice())?)?))?;
+                                    }
+                                    _ => {
+                                        write_scroller(unknown_cap_header, |w| Ok(write!(w, "name: {}, args cannot be displayed on Ledger", name_utf8)?))?;
+                                        set_from_thunk(destination, || Some(((), false))); // Fallback case
```

Comment:
@jonored i think recent developments you've made would allow us to handle this in a fully general way? namely, the interactive block protocol.

Reply:
Yep, but postponing that until there's a decent programming interface to it on the ledger side is reasonable.

Reply:
Although looking at the whole block, we sort of need some of this regardless; we're matching on particular recognized capabilities and showing better prompts for the ones we recognize. We could write this in an alternative-based way with the block protocol and retrying, though, which might be very good if a bit slower.